rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java

```diff
@@ -597,26 +600,23 @@ private void saveCoinbaseInformations() {
             safeSaveToRepository(getStorageKeyForCoinbaseInformation(blockHash), data, BridgeSerializationUtils::serializeCoinbaseInformation));
     }
 
-    public Optional<byte[]> getFastBridgeFederationScriptHash(Sha256Hash derivationArgsHash) {
+    public boolean isSavedFastBridgeFederationScriptHash(Sha256Hash derivationArgsHash) {
         if (!activations.isActive(RSKIP176)) {
-            return Optional.empty();
+            return false;
         }
+        byte[] data = repository.getStorageBytes(contractAddress, getStorageKeyForDerivationARguments(derivationArgsHash));
 
-        byte[] fastBridgeFedP2SH = repository.getStorageBytes(contractAddress, getStorageKeyForDerivationP2SHByHash(derivationArgsHash));
-
-        if (fastBridgeFedP2SH == null ) {
-            return Optional.empty();
-        }
-
-        return Optional.of(fastBridgeFedP2SH);
+        return ((data != null) && (data.length == 1) && (data[0] == FAST_BRIDGE_FEDERATION_SCRIPT_HASH_TRUE_VALUE));
     }
 
-    public void setDerivationArgumentsScriptHash(Sha256Hash derivationArgsHash, byte[] fastBridgeFedP2SH) {
+    public void setDerivationArgumentsScriptHash(Sha256Hash derivationArgsHash) {
```

Comment:
```suggestion
    public void markFastBridgeFederationDerivationHashAsUsed(Sha256Hash derivationArgsHash) {
```

Reply:
I think `setDerivationArgumentsScriptHash` is still there